autocompleter/widgets.py

```diff
@@ -30,3 +34,84 @@ def build_attrs(self, *args, **kwargs):
             'data-autocompleter-url': self.autocompleter_url
         })
         return attrs
+
+
+class AutocompleterSelectWidget(forms.MultiWidget):
+    """
+    This widget renders two adjacent <input> elements that provide a clean user interface
+    for searching via the Autocompleter API & selecting the desired object.
+
+    Wrapper for 2 widgets:
+        1. `AutocompleterWidget`: renders an <input> that acts as the search field.
+        2. `HiddenInput`: renders an <input> that holds the actual DB value (which is some kind of GUI)
+
+
+    :autocompleter_name - `str` specify the Autocompleter to use for search.
+    :display_name_field - `str` specify field from payload to display to user when result is selected.
+    :database_field     - `str` specify field from payload to save to the DB (should be a GUI).
+    :object_resolver    - `callable` that can fetch the object using the `database_field` value.
+    """
+
+    def __init__(self, autocompleter_name, autocompleter_url, display_name_field,
+                 database_field, object_resolver=None, *args, **kwargs):
+        self.autocompleter_name = autocompleter_name
+        self.display_name_field = display_name_field
+        self.database_field = database_field
+        self.object_resolver = object_resolver
+
+        widgets = [
+            AutocompleterWidget(autocompleter_url),
+            forms.HiddenInput()
+        ]
+        super().__init__(widgets, *args, **kwargs)
+
+    def decompress(self, value):
+        """
+        Decompress the field's DB value to both widgets <input> fields.
+
+            1. since `value` is the object identifier, we do a reverse lookup to fetch the AC
+               payload and show the `display_name_field` value.
+            2. `value` is held in the `HiddenInput` as the DB value.
+
+        returns [`display_name_field`, `db_value`]
+        """
+        # if DB field is empty, keep widgets blank.
+        if not value:
+            return [None, None]
+
+        # if no `object_resolver` callable is provided, we can fall back to showing
+        # the DB value in the search field. Not ideal, but better than leaving it blank.
+        if not self.object_resolver:
+            return ['ID: {}'.format(value), value]
+
+        # use the `object_resolver` to fetch the object instance.
+        try:
+            obj = self.object_resolver(value)
+        except ObjectDoesNotExist:
+            return ['Object does not exit', value]
+
+        # Use the `obj` and the Autocompleter to find the `ModelProvider`.
+        model_provider = self._get_model_provider(obj)
```

Comment:
1.don't you need to protect against `model_provider` being `None`
2. Don't we need to change the name of this variable to `provider`